tests/providers/elasticsearch/log/elasticmock/fake_elasticsearch.py

```diff
@@ -362,6 +363,18 @@ def match_must_phrase(document, matches, must):
                     # use in as a proxy for match_phrase
                     matches.append(document)
 
+    # Check index(es) exists
+    def _validate_search_targets(self, targets):
+        matches = set()
+        for target in targets:
+            if target == '_all':
+                matches.update(self.__documents_dict.keys())
+            elif '*' in target:
+                matches.update(fnmatch.filter(self.__documents_dict.keys(), target))
+            elif target not in self.__documents_dict:
+                raise NotFoundError(404, f'IndexMissingException[[{target}] missing]')
```

Comment:
The `keys()` call are not needed.

Since an exception is raised when `target` does not exist, maybe we should do the same if a wildcard pattern does not match anything?

```suggestion
            if target == '_all':
                matches.update(self.__documents_dict)
                continue
            target_matches = fnmatch.filter(self.__documents_dict, target)
            if not target_matches:
                raise NotFoundError(404, f'IndexMissingException[[{target}] missing]')
            matches.update(target_matches)
```

Reply:
@uranusjr actually Elasticsearch treats the case of a wildcard pattern not matching differently. See the `allow_no_indices` query parameter: https://www.elastic.co/guide/en/elasticsearch/reference/8.2/search-search.html#search-search-api-query-params
tl;dr if you specify an index name _without a pattern_ this results in 404. If you specify an index _with a pattern_ then by default you do not get a 404 even if there are no matching indices.

Reply:
For example:

    $ curl 'localhost:9200/whateverman-*/_search'
    {"took":0,"timed_out":false,"_shards":{"total":0,"successful":0,"skipped":0,"failed":0},"hits":{"total":{"value":0,"relation":"eq"},"max_score":0.0,"hits":[]}}

but

    $ curl 'localhost:9200/whateverman-foobar/_search'
    {"error":{"root_cause":[{"type":"index_not_found_exception","reason":"no such index [whateverman-foobar]","resource.type":"index_or_alias","resource.id":"whateverman-foobar","index_uuid":"_na_","index":"whateverman-foobar"}],"type":"index_not_found_exception","reason":"no such index [whateverman-foobar]","resource.type":"index_or_alias","resource.id":"whateverman-foobar","index_uuid":"_na_","index":"whateverman-foobar"},"status":404}

Reply:
However I haven't bothered to implement the case where`allow_no_indices=false` (would be necessary if the provider implementation was changed to override the default behavior):

    $ curl 'localhost:9200/whateverman-*/_search?allow_no_indices=false'
    {"error":{"root_cause":[{"type":"index_not_found_exception","reason":"no such index [whateverman-*]","resource.type":"index_or_alias","resource.id":"whateverman-*","index_uuid":"_na_","index":"whateverman-*"}],"type":"index_not_found_exception","reason":"no such index [whateverman-*]","resource.type":"index_or_alias","resource.id":"whateverman-*","index_uuid":"_na_","index":"whateverman-*"},"status":404}

Reply:
OK, let’s just fix the unnecessary `keys()` calls then.

Reply:
Done!